realm/realm-library/src/main/java/io/realm/internal/OsCollectionChangeSet.java

```diff
@@ -152,4 +153,39 @@ public String toString() {
         return string;
 
     }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public long getNativePtr() {
+        return nativePtr;
+    }
+
+    @Override
+    public long getNativeFinalizerPtr() {
+        return finalizerPtr;
+    }
+
+    // Returns the underlying error if an error was detected.
+    // The underlying layer will wrap it in an appropropriate exception class.
+    // `null` is returned if no error is present
+    @Nullable
+    private native Object nativeGetError(long nativePtr);
+
+    private native int nativeGetOldStatusCode(long nativePtr);
+
+    private native int nativeGetNewStatusCode(long nativePtr);
+
+    // Returns true if the data described by the subscription has been downloaded to the device,
+    // false if not. In either case, the query is run against the local dataset.
+    private native boolean nativeIsRemoteDataLoaded(long nativePtr);
```

Comment:
I might have misunderstood the part saying _query is run against the local dataset_ 
The name suggested a check to see if the data was loaded, I'm not sure why the query should run locally as well?

Reply:
The query only ever runs locally, this boolean indicates how "trustworthy" the result is. If data hasn't been synchronized with the server, you can get query results that would not be possible in a fully synchronized Realms.

Would renaming to `nativeIsRemoteDataSynchronized()` be a better name?